packages/docusaurus-migrate/src/index.ts

```diff
@@ -0,0 +1,707 @@
+        ),
+      );
+    } catch (errorInMigratingPages) {
+      console.log(
+        chalk.red(
+          `Error occurred while creating pages: ${errorInMigratingPages}`,
+        ),
+      );
+    }
+  } else {
+    try {
+      createDefaultLandingPage(newDir);
+      console.log(
+        chalk.green(
+          'Successfully created landing page (check migration page for more details)',
+        ),
+      );
+    } catch (errorInLandingPage) {
+      console.log(
+        chalk.red(
+          `Error occurred while creating landing page: ${errorInLandingPage}`,
+        ),
+      );
+    }
+  }
+
+  try {
+    migrateStaticFiles(siteDir, newDir);
+    console.log(chalk.green('Successfully migrated static folder'));
+  } catch (errorInStatic) {
+    console.log(
+      chalk.red(`Error occurred while copying static folder: ${errorInStatic}`),
+    );
+  }
+  try {
+    migrateBlogFiles(siteDir, newDir, classicPreset, shouldMigrateMdFiles);
+  } catch (errorInMigratingBlogs) {
+    console.log(
+      chalk.red(
+        `Error occurred while migrating blogs: ${errorInMigratingBlogs}`,
+      ),
+    );
+  }
+  try {
+    handleVersioning(siteDir, newDir, config, shouldMigrateMdFiles);
+  } catch (errorInVersion) {
+    console.log(
+      chalk.red(
+        `Error occurred while migrating versioned docs: ${errorInVersion}`,
+      ),
+    );
+  }
+
+  try {
+    migrateLatestDocs(siteDir, newDir, shouldMigrateMdFiles);
+  } catch (errorInDoc) {
+    chalk.red(`Error occurred while migrating docs: ${errorInDoc}`);
+  }
+
+  try {
+    migrateLatestSidebar(siteDir, newDir, classicPreset, siteConfig);
+  } catch (error) {
+    console.log(chalk.red(`Error occurred while migrating sidebar: ${error}`));
+  }
+
+  try {
+    fs.writeFileSync(
+      path.join(newDir, 'docusaurus.config.js'),
+      `module.exports=${JSON.stringify(config, null, 2)}`,
+    );
+    console.log(
+      chalk.green(
+        `Successfully created a new config file with new navbar and footer config`,
+      ),
+    );
+  } catch (error) {
+    console.log(
+      chalk.red(`Error occurred while creating config file: ${error}`),
+    );
+  }
+  try {
+    migratePackageFile(siteDir, deps, newDir);
+  } catch (error) {
+    console.log(
+      chalk.red(
+        `Error occurred while creating package.json file for project: ${error}`,
+      ),
+    );
+  }
+  console.log('Completed migration from v1 to v2');
+}
+
+export function createConfigFile(
+  siteConfig: VersionOneConfig,
+): VersionTwoConfig {
+  const homePageId = siteConfig.headerLinks?.filter((value) => value.doc)[0]
+    .doc;
+
+  const customConfigFields: Record<string, any> = {};
+  // add fields that are unknown to v2 to customConfigFields
+  Object.keys(siteConfig).forEach((key: any) => {
+    const knownFields = [
+      'title',
+      'tagline',
+      'url',
+      'baseUrl',
+      'organizationName',
+      'projectName',
+      'scripts',
+      'stylesheets',
+      'favicon',
```

Comment:
migrating v1 site leads to this. We should filter much more fields here, like the known theme attributes

![image](https://user-images.githubusercontent.com/749374/88296052-81d1ef00-ccfe-11ea-9fb2-6f9ef7bed3f7.png)

Reply:
we have added all the available theme attributes even twitter link is added to the footer

Reply:
Yes, the v1 theme attributes have been put in the appropriate v2 theme config.

But these known v1 theme attributes shouldn't end up too in the v2 config.customFields, as they are now duplicated in the final config.

The final v2 config should barely have more than 2 or 3 custom fields.

Look at D2 website, it only has one, yet it was migrated a few years ago from v1 site, so migrating automatically v1 site with the cli should rather give us a quite similar output.

![image](https://user-images.githubusercontent.com/749374/88308321-307d2c00-cd0d-11ea-92b9-5d36abaca9e4.png)

Reply:
I think the remaining fields that could be useful are:
- translationRecruitingLink
- gaTrackingId
- facebookAppId
- twitterUsername 
- highlight.theme
- scrollToTopOptions

The other fields are either already included in config or deprecated